moped-etl/arcgis/utils.py

```diff
@@ -64,6 +65,33 @@ def make_hasura_request(*, query, variables=None):
         raise ValueError(data)
 
 
+def make_hasura_sql_query(*, query, variables=None):
+    """Fetch data from hasura
+
+    Args:
+        query (str): the hasura query
+
+    Raises:
+        ValueError: If no data is returned
+
+    Returns:
+        dict: Hasura JSON response data
+    """
+    headers = {
+        "X-Hasura-Admin-Secret": HASURA_ADMIN_SECRET,
+        "Content-Type": "application/json",
+    }
+    payload = {"type": "run_sql", "args": {"sql": query}}
+
+    res = requests.post(HASURA_SQL_ENDPOINT, json=payload, headers=headers)
+    res.raise_for_status()
+    data = res.json()
+    try:
+        return data["result"]
+    except KeyError:
+        raise ValueError(data)
```

Comment:
New sibling function to 'make_hasura_request()` that takes SQL and executes it and returns the result. Makes me wonder what secures this from people writing arbitrary queries to it with a token extracted from the moped web app. I hope it's restricted to having the `HASURA_ADMIN_SECRET`. But anyway, that's entirely out of the scope of this issue.